protocol-definitions/Jet.yaml

```diff
@@ -321,3 +321,22 @@ methods:
           nullable: false
           since: 2.6
           doc: ''
+  - id: 17
+    name: setJobConfig
+    since: 2.6
+    doc: ''
+    request:
+      retryable: true
+      partitionIdentifier: -1
+      params:
+        - name: jobId
+          type: long
+          nullable: false
+          since: 2.6
+          doc: ''
+        - name: config
+          type: Data
+          nullable: false
+          since: 2.6
+          doc: ''
```

Comment:
Shouldn't we rather introduce a custom type which can be evolved? Otherwise we end up with a binary blob that can't be changed.

Reply:
@hazelcast/apisjava This is in line with `submitJob`, which specifies `jobConfig` as `Data`. How should we proceed? What is the benefit of introducing custom types?

Reply:
The rule of thumb is that, if the parameter is of a known type we should use a custom type to represent it. The reasoning is what Frantisek wrote. If we make it data, it means it can break backward compatibility when the identified data representation of the parameter changes. 

However, I believe those other data parameters were used because Jet doesn't have the concept of backward compatibility. Do we allow old/new clients to submit jobs to new/old members?

Reply:
No, we don't allow that. A client must have the same minor version as the member to which it is connected, so members will always be able to deserialize `JobConfig`s sent by clients.

Reply:
Yes, to use Jet one has to use the exact same minor version. Otherwise many other things will break.
SQL avoids this problem, you can change the job config with a SQL command.